scipy/stats/_distn_infrastructure.py

```diff
@@ -2164,16 +2176,39 @@ def restore(args, theta):
 
             def func(theta, x):
                 newtheta = restore(args[:], theta)
-                return self._penalized_nnlf(newtheta, x)
+                return objective(newtheta, x)
 
         return x0, func, restore, args
 
+    def _moment_error_norm(self, theta, x):
+        loc, scale, args = self._unpack_loc_scale(theta)
+        if not self._argcheck(*args) or scale <= 0:
+            return inf
+
+        n = len(theta) - len(self._fixedn)
+        dist_moments = np.array([self.moment(i+1, *args, loc=loc, scale=scale)
+                                 for i in range(n)])
+        if not np.all(np.isfinite(dist_moments)):
+            raise ValueError("Method of moments encountered a non-finite "
+                             "distribution moment and cannot continue. "
+                             "Consider trying method='MLE'.")
+
+        if self._data_moments is None:
+            exponents = (np.arange(1, n+1))[:, np.newaxis]
+            self._data_moments = np.sum(x[np.newaxis, :]**exponents/len(x),
+                                        axis=1)
```

Comment:
Can't use `scipy.stats.moment` directly because that's for calculating the sample moments about the sample mean. Would be nice to add an option to that function to calculate about another point.